LogicLayer/Groups/IGroupsService.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using GameBoard.LogicLayer.Groups.Dtos;
+using JetBrains.Annotations;
+
+namespace GameBoard.LogicLayer.Groups
+{
+    public interface IGroupsService
+    {
+        // Gets groups created by this user(userName).
+        [NotNull]
+        [ItemNotNull]
+        Task<IEnumerable<GroupDto>> GetGroupsByUserNameAsync([NotNull] string userName);
+
+        Task<GroupDto> GetGroupByNamesAsync([NotNull] string owner, [NotNull] string groupName);
```

Comment:
Mark this with `[ItemNotNull]`. Also this one probably could use a comment stating that it throws an `InvalidOperationException` if the group does not exist, since basically all other `GetX` methods we have return `null` in that scenario.

Reply:
Marking as done.